bus/client.go

```diff
@@ -787,6 +788,70 @@ func (c *Client) AddUploadingSector(ctx context.Context, uID api.UploadID, id ty
 	return
 }
 
+func (c *Client) AddPartialSlab(ctx context.Context, data []byte, minShards, totalShards uint8, contractSet string) (slabs []object.PartialSlab, err error) {
+	c.c.Custom("POST", "/slabs/partial", nil, &api.AddPartialSlabResponse{})
+	values := url.Values{}
+	values.Set("minShards", fmt.Sprint(minShards))
+	values.Set("totalShards", fmt.Sprint(totalShards))
+	values.Set("contractSet", contractSet)
+
+	u, err := url.Parse(fmt.Sprintf("%v/slabs/partial", c.c.BaseURL))
+	if err != nil {
+		panic(err)
+	}
+	u.RawQuery = values.Encode()
+	req, err := http.NewRequestWithContext(ctx, "POST", u.String(), bytes.NewReader(data))
+	if err != nil {
+		panic(err)
+	}
+	req.SetBasicAuth("", c.c.WithContext(ctx).Password)
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer io.Copy(io.Discard, resp.Body)
+	defer resp.Body.Close()
+	if resp.StatusCode != 200 {
+		err, _ := io.ReadAll(resp.Body)
+		return nil, errors.New(string(err))
+	}
+	var apsr api.AddPartialSlabResponse
+	err = json.NewDecoder(resp.Body).Decode(&apsr)
+	if err != nil {
+		return nil, err
+	}
+	return apsr.Slabs, nil
+}
+
+func (c *Client) FetchPartialSlab(ctx context.Context, key object.EncryptionKey, offset, length uint32) ([]byte, error) {
+	c.c.Custom("GET", fmt.Sprintf("/slabs/partial/%s", key), nil, &[]byte{})
+	values := url.Values{}
+	values.Set("offset", fmt.Sprint(offset))
+	values.Set("length", fmt.Sprint(length))
+
+	u, err := url.Parse(fmt.Sprintf("%s/slabs/partial/%s", c.c.BaseURL, key))
+	if err != nil {
+		panic(err)
+	}
+	u.RawQuery = values.Encode()
+	req, err := http.NewRequestWithContext(ctx, "GET", u.String(), nil)
+	if err != nil {
+		panic(err)
+	}
+	req.SetBasicAuth("", c.c.WithContext(ctx).Password)
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer io.Copy(io.Discard, resp.Body)
+	defer resp.Body.Close()
+	if resp.StatusCode != 200 && resp.StatusCode != 206 {
```

Comment:
is it possible that we receive a 204 here if we mess up the `offset` somehow?

Reply:
No, it shouldn't. If we mess up offset and length there would be some EOF error.